windows/hidapi_descriptor_reconstruct.h

```diff
@@ -125,6 +126,9 @@ typedef struct hid_pp_link_collection_node_ {
 	// Same as the public API structure HIDP_LINK_COLLECTION_NODE, but without PVOID UserContext at the end
 } hid_pp_link_collection_node, *phid_pp_link_collection_node;
 
+static_assert(sizeof(struct hid_pp_link_collection_node_) == 16,
+    "Size of struct hid_pp_link_collection_node_ not as expected. This might break binary compatibility");
```

Comment:
as much as I like this - there is two problems with this part:
1) this is a half-measure - if we're to static_asset it - needs to be done for all similar structs in this file
2) `static_asset` is a C++ feature, in C - it is a [bit different](https://en.cppreference.com/w/c/language/_Static_assert), as always; need to confirm if all compilers support it equally

Reply:
1) Yes it's not perfect, but it is more than nothing. And this covers `ULONG`, `USHORT`, `USAGE` and `ULONG`-bit-fields.
2) Actually it is part of C++11 [cppref](https://en.cppreference.com/w/c/language/_Static_assert) / [MS](https://learn.microsoft.com/en-us/cpp/c-language/static-assert-c?view=msvc-170)

Reply:
Hm, you got me there, I should be pay more attention to all of the wording.
I'm generally skeptic about C11 support by the compilers (in opposed to C++11), i.e. threading as introduced by C11 is not yet supported by all of the compilers...

> more than nothing

OK, lets have it.

But I'm pretty sure you're not considering that in addition to do  "more than nothing (good)" it also brings some negative side-effect, e.g. when a person sees there is a single `static_assert` it make assume one of:
- there is a similar `static_assert` for all structs for which the size has to be enforced (that is not true);
- structs that don't have it - do not have size restrictions and may be changed for convenience (that is not true either);
I've seen similar cases in my practice.

Reply:
Nothing that cannot be fixed with a comment. ;)

Reply:
Right, becase:
- all developers read comments;
- compilers read comments;

(: